server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -185,7 +184,7 @@ public T parse(String name, Map<String, Object> node, ParserContext parserContex
         protected QueryProcessor geometryQueryBuilder;
 
         protected AbstractGeometryFieldType(String name, boolean indexed, boolean hasDocValues, Map<String, String> meta) {
-            super(name, indexed, hasDocValues, meta);
+            super(name, indexed, hasDocValues, TextSearchInfo.GEOMETRY, meta);
```

Comment:
This is a bit weird, but geometry fields currently let you go all the way through to calling `termQuery` before throwing an error, so this preserves the current behaviour.  It might allow us to support match queries against geometry fields via WKT or something similar in the future, so I think we should leave it in for now.